### DISCUSSION TITLE
支援的裝置 / Supported devices

### DISCUSSION THREAD

**Discussion by osk2** [osk2/panasonic_smart_app#42] [Announcements]
下列裝置經過測試確定可支援，若你的裝置測試後確定支援也可以在下方回報裝置型號

若為其他問題回報請移駕至 [Issues](https://github.com/osk2/panasonic_smart_app/issues)

## 除濕機
| 裝置型號   | 模組型號 |
| ---------- | -------- |
| F-Y28EX    | CZ-T006 |
| F-Y24GX   | CZ-T006 |
| F-Y20JH    | _(內建)_ |
| F-Y26JH    | _(內建)_ |
| F-Y32JH    | _(內建)_ |
| F-YV32LX  | _(內建)_ |
| F-YV50LX  | _(內建)_ |
| F-YV32NXK | _(內建)_ |
| F-YV32MH | _(內建)_ |
|F-YD64LX| _(內建)_ |

## 冷氣
| 裝置型號   | 模組型號 |
| ---------- | -------- |
| CS-PX22FA2 | CZ-T007  |
| CS-PX28FA2 | CZ-T007  |
| CS-PX36FA2 | CZ-T007  |
| CS-PX50FA2 | CZ-T007  |
| CS-PX63FA2 | CZ-T007  |
| CS-PX80FA2 | CZ-T007 | 
| CS-PX36JA2 | _(內建)_ |
| CS-PX50JA2 | _(內建)_ |
| CS-PX63JA2 | _(內建)_ |
| CS-QX28FA2 | CZ-T007 |
| CS-QX40FA2 | CZ-T007 |
| CS-QX71FA2 | CZ-T007  |
| CS-LX36A2   | CZ-T007  |
| CS-LJ22BA2 | CZ-T007  |
| CS-RX22JA2  | _(內建)_ |
| CS-RX28JA2  | _(內建)_ |
| CS-RX40JA2  | _(內建)_ |
| CS-RX22NA2 | _(內建)_ |
| CS-RX28NA2 | _(內建)_ |
| CS-RX71NA2  | _(內建)_ |
| CS-RX28GA2 | _(內建)_ |
| CS-RX36GA2 | _(內建)_ |
| CS-RX50GA2 | _(內建)_ |
| CS-RX63GA2 | _(內建)_ |
| CS-RX71GA2  | _(內建)_ |
| CS-RX90GA2  | _(內建)_ |
| CS-UX22BA2  | _(內建)_ |
| CS-UX28BA2  | _(內建)_ |
| CS-UX36BA2  | _(內建)_ |

## 洗衣機
| 裝置型號   | 模組型號 |
| ---------- | -------- |
|NA-V160HDH|_(內建)_|
|NA-V150MDH|_(內建)_|

##  全熱交換器
| 裝置型號   | 模組型號 |
| ---------- | -------- |
|FY-15ZY1W|CZ-T012|

## 冰箱
| 裝置型號   | 模組型號 |
| ---------- | -------- |
|NR-D615XGS|_(內建)_|

---
#### (English version)

The following devices were tested. Feel free to report your device model.

## Dehumidifier

| Device Model | Module Type  |
| ------------ | ----------- |
| F-Y28EX      | CZ-T006     |
| F-Y24GX      | CZ-T006     |
| F-Y20JH      | _(Built-in)_ |
| F-Y26JH      | _(Built-in)_ |
| F-Y32JH      | _(Built-in)_ |
| F-YV32LX    | _(Built-in)_ |
| F-YV50LX    | _(Built-in)_ |
| F-YV32MH   | _(Built-in)_ |
| F-YV32NXK  | _(Built-in)_ |
| F-YD64LX    | _(Built-in)_ |

## Air Conditioner
| Device Model | Module Type  |
| ------------ | ------------ |
| CS-PX22FA2   | CZ-T007      |
| CS-PX28FA2   | CZ-T007      |
| CS-PX36FA2   | CZ-T007      |
| CS-PX50FA2   | CZ-T007      |
| CS-PX63FA2   | CZ-T007      |
| CS-PX80FA2   | CZ-T007      |
| CS-PX36JA2   | _(Built-in)_ |
| CS-PX50JA2   | _(Built-in)_ |
| CS-PX63JA2   | _(Built-in)_ |
| CS-QX28FA2  | CZ-T007      |
| CS-QX40FA2  | CZ-T007      |
| CS-QX71FA2   | CZ-T007      |
| CS-LX36A2     | CZ-T007      |
| CS-LJ22BA2    | CZ-T007     |
| CS-RX22JA2    | _(Built-in)_ |
| CS-RX28JA2    | _(Built-in)_ |
| CS-RX40JA2    | _(Built-in)_ |
| CS-RX22NA2   | _(Built-in)_ |
| CS-RX28NA2   | _(Built-in)_ |
| CS-RX71NA2    | _(Built-in)_ |
| CS-RX28GA2   | _(Built-in)_ |
| CS-RX36GA2   | _(Built-in)_ |
| CS-RX50GA2   | _(Built-in)_ |
| CS-RX63GA2   | _(Built-in)_ |
| CS-RX71GA2   | _(Built-in)_ |
| CS-RX90GA2  | _(Built-in)_ |
| CS-UX22BA2  | _(Built-in)_ |
| CS-UX28BA2  | _(Built-in)_ |
| CS-UX36BA2  | _(Built-in)_ |

## Washing Machine
| Device Model | Module Type  |
| ---------- | -------- |
|NA-V160HDH|_(Built-in)_ |
|NA-V150MDH|_(Built-in)_ |

## ERV
| Device Model | Module Type  |
| ---------- | -------- |
|FY-15ZY1W|CZ-T012|

## Refrigerator
| Device Model | Module Type  |
| ---------- | -------- |
|NR-D615XGS|_(Built-in)_ |

**Comment by ayhaadam** [comment]
回報，除濕機 F-Y20JH 可用，模組為內建

**Reply by osk2** [reply]
感謝回報，已更新資訊

**Reply by xxzxzqs997** [reply]
你好，我明明输入的账号密码是对的，可是总是提示错误登录不了这个问题请问有什么很好的解决办法吗
![Uploading IMG_5342.png…]()

**Comment by guoer11** [comment]
昨天才安裝cz-t007
回報CS-LX36A2冷氣可以使用，感謝

**Reply by osk2** [reply]
哇沒看過這個型號，已經更新上去了 👍🏼

**Reply by xxzxzqs997** [reply]
你好，我明明输入的账号密码是对的，可是总是提示错误登录不了这个问题请问有什么好的解决办法吗
![Uploading IMG_5342.png…]()

**Comment by gaozhitw** [comment]
回報：
CS-RX22JA2
CS-RX28JA2

可用，模組內建

**Reply by osk2** [reply]
已更新 👍

**Comment by homeleo1818** [comment]
Hi 你好 我已經安裝好HACS並透過它下載Panasonic Smart App，但下載完成沒有跳出任何視窗可進行下一動如輸入帳號密碼
請問這可能是什麼問題?有遇過嗎?

**Reply by osk2** [reply]
在 HACS 下載後還要再到 HA 的設定裡面新增整合，搜尋並新增 Panasonic Smart App，這樣才是真的完成安裝

**Comment by homeleo1818** [comment]
Hi osk2:
             感謝你的撥空回復，我已經能在設定=>裝置與服務裡面看到HACS以及也安裝了Smart App，
             安裝後看到Pending Restart，手動重啟HA，目前仍無法看到視窗自動跳出，相當疑惑哪裡
             還需要設定些什麼嗎?煩請協助解惑，謝謝

osk2 ***@***.***> 於 2022年5月13日 週五 下午5:29寫道：

**Reply by osk2** [reply]
聽你的敘述應該只用HACS把套件載下來而已，還少安裝的步驟：
1. 到HA設定-›裝置與服務，按右下的新增整合
2. 搜尋 Panasonic Smart App
3. 開始安裝時HA就會自動跳出設定視窗了

**Comment by homeleo1818** [comment]
原來如此，感謝解惑啊，搞了兩天，都不知道原因，原來要在HA再新增整合一次才會出現
現在總算能看到家裡的設備，我在研究導入HK，謝謝大大

osk2 ***@***.***> 於 2022年5月14日 週六 上午1:04寫道：

**Reply by homeleo1818** [reply]
另外回報CS-RX63GA2可正常連線使用，模組內建

**Reply by osk2** [reply]
感謝，已更新

**Comment by jasonlai79** [comment]
CS-LJ22BA2 搭配CZ-T007
測試可用，但是整合裡面只有感測器沒有其他開關在裝置裡面，要額外找，希望能整合到一起！謝謝🙏

_2022/08/19 編輯：隱藏圖片以節省空間_

**Reply by osk2** [reply]
已更新
我的HA也有這個問題，還以為只有我自己遇到，我再找時間看看是哪裡有問題

**Comment by allanXue0805** [comment]
CS-RX40JA2可用，模組內建。感謝!!!

**Reply by osk2** [reply]
感謝回報，已經更新了

**Comment by ivis0516** [comment]
謝謝您的分享，CS-PX80FA2  這台在HA裡面看得到，但是homekit裡面就是無法顯示，請問有什麼設定要做嗎？

**Reply by ivis0516** [reply]
後來重裝一次就設定好了，但是我發現用了這個到homekit後，原廠程式裡面就顯示一片空白，大家的也都是這樣嗎？

**Reply by osk2** [reply]
應是超量使用的問題，詳情請見：https://github.com/osk2/panasonic_smart_app/discussions/31
另外，若使用上有任何異常可以開一個新的 [issue](https://github.com/osk2/panasonic_smart_app/issues)，這邊就單純給大家回報可用裝置 🙏

**Reply by ivis0516** [reply]
CS-PX80FA2 可用！謝謝您

**Reply by osk2** [reply]
模組型號是 CZ-T007 嗎？

**Comment by yingzeliao** [comment]
CS-RX90GA2可用, 模組內建, 感謝

**Comment by ivis0516** [comment]
對

從我的iPhone傳送

**Comment by huaxiac4** [comment]
我來自大陸，冰箱型號：NR-EC26WPA-W 應該是僅限大陸販售的型號，請問是否支援？

**Reply by osk2** [reply]
支援與否是透過搭配的 app 決定的，Panasonic Smart App 僅限台灣使用，所以我想除台灣以外的裝置都無法支援

**Comment by cenwy** [comment]
F-Y32JH 只抓得都感測器，沒有可控開關
![image](https://user-images.githubusercontent.com/118907463/207263157-7c1e2800-df02-4f5e-b7ad-6322f12fec79.png)

**Reply by osk2** [reply]
這問題之前已經有人回報過了
Ref: https://github.com/osk2/panasonic_smart_app/issues/59

**Comment by msg00235** [comment]
洗衣機NA-V160HDH可用 模組內建

**Comment by ductu3590** [comment]
Please support model: CS-YZ18UKH-8

**Reply by osk2** [reply]
I'm afraid we can't do that, we only support devices that can be added to Panasonic Smart App.

**Comment by btons** [comment]
洗衣機NA-V150MDH 可用 模組內建

**Comment by et952035** [comment]
請問大大這可以加入Homekit嗎?
拜託教教我，謝謝

**Reply by osk2** [reply]
嗨，建議你到 [Home Assistant 社團](https://www.facebook.com/groups/151166072456061) 發文會比較多人幫忙，這邊只提供 Panasonic 套件的支援

**Comment by joe-hsiao** [comment]
回報：
CS-RX22NA2
CS-RX28NA2
CS-RX71NA2
內建模組可用。

**Comment by ghost** [comment]
洗衣機 NA-LX128有出現實體 但不可用 謝謝
![image](https://github.com/osk2/panasonic_smart_app/assets/96696890/9bc4c556-f317-49db-80f6-5ca3ad228a2e)

**Reply by osk2** [reply]
顯示不可用時其他裝置都正常嗎？

**Reply by ghost** [reply]
我只有使用一個Panasonic 家電 所以沒有其他裝置
其他整合內的實體都可以正常使用 謝謝

**Reply by hSATAC** [reply]
卡，請問 LX128 現在有支援了嗎？

**Comment by Adios** [comment]
去年最新的除濕機 LX 系列的 F-YV32LX 可以讀到數值，應該正常

**Comment by qqq1236** [comment]
冷氣吊隱CS-J63BDA2 可加入實體，但全都不可用
![IMG_1280](https://github.com/osk2/panasonic_smart_app/assets/101200808/de9d8436-c734-46a7-9eda-fb86bb1056b3)

**Comment by Zuiikin** [comment]
> 以下装置经过测试确定可以支持，你也可以在下方返回你的装置型号
> 
> ## 除湿机
> 装置型号	模组型号
> F-Y28EX	CZ-T006
> F-Y24GX	CZ-T006
> F-Y20JH	_(内建)_
> F-Y26JH	_(内建)_
> F-YV32LX	_(内建)_
> ## 冷气
> 装置型号	模组型号
> CS-PX22FA2	CZ-T007
> CS-PX28FA2	CZ-T007
> CS-PX36FA2	CZ-T007
> CS-PX50FA2	CZ-T007
> CS-PX63FA2	CZ-T007
> CS-PX80FA2	CZ-T007
> CS-QX28FA2	CZ-T007
> CS-QX40FA2	CZ-T007
> CS-QX71FA2	CZ-T007
> CS-LX36A2	CZ-T007
> CS-LJ22BA2	CZ-T007
> CS-RX22JA2	_(内建)_
> CS-RX28JA2	_(内建)_
> CS-RX40JA2	_(内建)_
> CS-RX22NA2	_(内建)_
> CS-RX28NA2	_(内建)_
> CS-RX71NA2	_(内建)_
> CS-RX28GA2	_(内建)_
> CS-RX36GA2	_(内建)_
> CS-RX50GA2	_(内建)_
> CS-RX63GA2	_(内建)_
> CS-RX71GA2	_(内建)_
> CS-RX90GA2	_(内建)_
> ## 洗衣机
> 装置型号	模组型号
> NA-V160HDH	_(内建)_
> NA-V150MDH	_(内建)_
> #### （英文版）
> 测试了以下设备。请随时报告您的设备型号。
> 
> ## 除湿机
> 设备型号	模块类型
> F-Y28EX	CZ-T006
> F-Y24GX	CZ-T006
> F-Y20JH	_（内置）_
> F-Y26JH	_（内置）_
> F-YV32LX	_（内置）_
> ## 冷气机
> 设备型号	模块类型
> CS-PX22FA2	CZ-T007
> CS-PX28FA2	CZ-T007
> CS-PX36FA2	CZ-T007
> CS-PX50FA2	CZ-T007
> CS-PX63FA2	CZ-T007
> CS-PX80FA2	CZ-T007
> CS-QX28FA2	CZ-T007
> CS-QX40FA2	CZ-T007
> CS-QX71FA2	CZ-T007
> CS-LX36A2	CZ-T007
> CS-LJ22BA2	CZ-T007
> CS-RX22JA2	_（内置）_
> CS-RX28JA2	_（内置）_
> CS-RX40JA2	_（内置）_
> CS-RX22NA2	_（内置）_
> CS-RX28NA2	_（内置）_
> CS-RX71NA2	_（内置）_
> CS-RX28GA2	_（内置）_
> CS-RX36GA2	_（内置）_
> CS-RX50GA2	_（内置）_
> CS-RX63GA2	_（内置）_
> CS-RX71GA2	_（内置）_
> CS-RX90GA2	_（内置）_
> ## 洗衣机
> 设备型号	模块类型
> NA-V160HDH	_（内置）_
> NA-V150MDH	_（内置）_



> 以下装置经过测试确定可以支持，你也可以在下方返回你的装置型号
> 
> ## 除湿机
> 装置型号	模组型号
> F-Y28EX	CZ-T006
> F-Y24GX	CZ-T006
> F-Y20JH	_(内建)_
> F-Y26JH	_(内建)_
> F-YV32LX	_(内建)_
> ## 冷气
> 装置型号	模组型号
> CS-PX22FA2	CZ-T007
> CS-PX28FA2	CZ-T007
> CS-PX36FA2	CZ-T007
> CS-PX50FA2	CZ-T007
> CS-PX63FA2	CZ-T007
> CS-PX80FA2	CZ-T007
> CS-QX28FA2	CZ-T007
> CS-QX40FA2	CZ-T007
> CS-QX71FA2	CZ-T007
> CS-LX36A2	CZ-T007
> CS-LJ22BA2	CZ-T007
> CS-RX22JA2	_(内建)_
> CS-RX28JA2	_(内建)_
> CS-RX40JA2	_(内建)_
> CS-RX22NA2	_(内建)_
> CS-RX28NA2	_(内建)_
> CS-RX71NA2	_(内建)_
> CS-RX28GA2	_(内建)_
> CS-RX36GA2	_(内建)_
> CS-RX50GA2	_(内建)_
> CS-RX63GA2	_(内建)_
> CS-RX71GA2	_(内建)_
> CS-RX90GA2	_(内建)_
> ## 洗衣机
> 装置型号	模组型号
> NA-V160HDH	_(内建)_
> NA-V150MDH	_(内建)_
> #### （英文版）
> 测试了以下设备。请随时报告您的设备型号。
> 
> ## 除湿机
> 设备型号	模块类型
> F-Y28EX	CZ-T006
> F-Y24GX	CZ-T006
> F-Y20JH	_（内置）_
> F-Y26JH	_（内置）_
> F-YV32LX	_（内置）_
> ## 冷气机
> 设备型号	模块类型
> CS-PX22FA2	CZ-T007
> CS-PX28FA2	CZ-T007
> CS-PX36FA2	CZ-T007
> CS-PX50FA2	CZ-T007
> CS-PX63FA2	CZ-T007
> CS-PX80FA2	CZ-T007
> CS-QX28FA2	CZ-T007
> CS-QX40FA2	CZ-T007
> CS-QX71FA2	CZ-T007
> CS-LX36A2	CZ-T007
> CS-LJ22BA2	CZ-T007
> CS-RX22JA2	_（内置）_
> CS-RX28JA2	_（内置）_
> CS-RX40JA2	_（内置）_
> CS-RX22NA2	_（内置）_
> CS-RX28NA2	_（内置）_
> CS-RX71NA2	_（内置）_
> CS-RX28GA2	_（内置）_
> CS-RX36GA2	_（内置）_
> CS-RX50GA2	_（内置）_
> CS-RX63GA2	_（内置）_
> CS-RX71GA2	_（内置）_
> CS-RX90GA2	_（内置）_
> ## 洗衣机
> 设备型号	模块类型
> NA-V160HDH	_（内置）_
> NA-V150MDH	_
添加以下空调控制吧

**Comment by GodricBai** [comment]
目前F-YV38LX無法讀取並控制
<img width="1051" alt="image" src="https://github.com/osk2/panasonic_smart_app/assets/141562821/0838cd03-09e2-461f-9a91-11c7b5064d00">

**Comment by 30billionretire** [comment]
CS-SX36DBA2 內建wifi :
1. 有時候會可以開啟冷氣但無法關閉
2. 有時候會可以關閉但無法開啟冷氣
3. 但操作時冷氣都會有反應只是不會是預想中的樣子（開or關）

**Reply by osk2** [reply]
這邊有幾個冷氣開關狀態相反的回報，但我自己沒有遇到，若有人能提供 log 的話才有辦法解決
https://github.com/osk2/panasonic_smart_app/issues/103

**Reply by 30billionretire** [reply]
更新:  後來看到issue 中有人更新至2.8.1後有所解決，便也嘗試更新，更新後目前使用都已正常了！

**Comment by sahsu** [comment]
F-YV50LX 也可以直接用 ( https://www.panasonic.com/tw/consumer/air-conditioner/dehumidifier/f-yv50lx.html )
<img width="904" alt="image" src="https://github.com/osk2/panasonic_smart_app/assets/1015931/7176dabd-dd60-41e6-b79d-33575e95071f">

**Comment by samfeng0517** [comment]
[F-YV32MH](https://www.panasonic.com/tw/consumer/air-conditioner/dehumidifier/f-yv32mh.html) PM2.5 沒有實體，除此之外皆正常

![螢幕擷取畫面 2024-06-11 005058](https://github.com/osk2/panasonic_smart_app/assets/66020451/a250383d-9698-4223-bfe4-0b524e737c5a)

**Comment by xxzxzqs997** [comment]
你好，我明明输入的账号密码是对的，可是总是提示错误登录不了这个问题请问有什么好的解决办法吗？求大神教一下
![Uploading IMG_5342.png…]()

**Reply by osk2** [reply]
參考 https://github.com/osk2/panasonic_smart_app/issues/107

**Comment by JyunWei-Su** [comment]
回報 UX系列冷氣機可用 (CS-UX28BA2、CS-UX36BA2)
![image](https://github.com/user-attachments/assets/b1b3f8fe-fc86-4b98-a11d-6432c5b50d7b)

**Comment by chun950yi** [comment]
大大您好，想詢問這邊沒有提供空氣清淨機的支援型號，是只要能連上App的都有支援嗎？
近期想買F-P60LH的空氣清淨機，請問有支援嗎

**Reply by osk2** [reply]
基本上能加入 app 的裝置都有支援
這邊只是將網友確定有完整支援的裝置列出來，不代表沒列出的無法使用

**Comment by kulaLin** [comment]
換新冷氣回報，CS-UX22BA2可支援
內建模組
![image](https://github.com/user-attachments/assets/8e8333b7-4935-4c14-8a9b-49be655359ed)

**Comment by blackmeat1225** [comment]
NA-V150MDH 不可用, 之前可以
![NA-V150MDH](https://github.com/user-attachments/assets/fdd42137-9131-466e-8305-2d1a1d2665ba)

**Comment by Darric-Lu** [comment]
CS-PX63JA2 內建 wifi
CS-PX50JA2 內建 wifi 
CS-PX36JA2 內建 wifi 

皆可以使用

![CS-PX63JA2](https://github.com/user-attachments/assets/e53701a3-dbc2-4a61-b2af-38391a7ef596)

![CS-PX50JA2](https://github.com/user-attachments/assets/b509843b-8173-4195-b7d0-b2a5e62fa529)

![CS-PX36JA2](https://github.com/user-attachments/assets/a542f7fc-d266-42dc-a9cd-6596c7d30f2b)

**Comment by k4l6u0** [comment]
回報:
落地形除濕機F-Y32JH : 可登入使用, 內建WiFi模組, 通常可取得濕度, 但常常會有’Unavailable'的數據,
目前看到有如截圖
![image](https://github.com/user-attachments/assets/2658811f-2a3f-499a-aa51-4004002c0615)

**Reply by k4l6u0** [reply]
另外, 過陣子我有機會增加以下設備, 屆時可以協助測試.

全戶除濕機F-YD64LX
全熱交換機FY-25ZY1W
冰箱NR-D615XGS

**Reply by osk2** [reply]
看起來除濕機有更名過，那些 unavailable 都是改名前留下的實體，重新安裝可以解決

**Reply by acchiang2000** [reply]
家有FY-25ZY1W全熱交換器和日製冰箱NR-F607HX-W1, 使用HA panasonic_smart_app 2.11.0
兩個裝置都可以接入HA
但是冰箱的:
冷藏溫度, 冷凍溫度, 微凍結溫度
三個實體無法使用, 其他部分則是正常
必須手動關閉那三個無法使用的實體, 才能使得全熱交換器FY-25ZY1W正常使用, 不然會顯示冰箱實體溫度相關的錯誤訊息

**Comment by k4l6u0** [comment]
目前冰箱+吊隱除濕機都可以正常顯示
全熱因為還沒安裝會wifi模組，所以還未串接

[Image.png]

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by a088753349** [comment]
回報，最新的除濕機 F-YV32NXK 可用

**Comment by hsiu-chan** [comment]
![image](https://github.com/user-attachments/assets/4d5496ff-d9a0-41bc-b9e7-e615fd358796)
回報，CS-UJ40BA2使用CZT007，冷氣不能控制溫度

**Reply by osk2** [reply]
UJ系列請見 #138

**Comment by k4l6u0** [comment]
![image](https://github.com/user-attachments/assets/28952cb7-e1d6-4347-92cb-2284eb200326)
F-YD64LX吊隱式除濕機的cost統計看起來怪怪的
entity顯示是"dehumidifierchu_shi_ji_ben_yue_hao_dian_liang_cost", 那應該是本月到現在累積的耗電量
但是圖上看起來是每天歸零, 也不是累加上去.

如果有需要抓取什麼資料, 再請告訴我

**Comment by zyi840449** [comment]
洗衣機NA-V150RPH可連線，能看到以下6個實體狀態:
本月耗電量、本月碳排放、目前行程、目前模式、剩餘洗衣時間、運轉狀態

但洗衣機關機後，只剩下本月碳排放看的到數字，其他實體都會變成"不可用"狀態
而且系統的日誌會一直出現錯誤訊息

記錄器: homeassistant
來源: custom_components/panasonic_smart_app/sensor.py:576
整合: Panasonic Smart App ([相關文件](https://github.com/osk2/panasonic_smart_app), [問題](https://github.com/osk2/panasonic_smart_app/issues))
首次發生: 上午12:52:15 (154 次出現)
最後紀錄: 上午8:48:17

Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 268, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 479, in _async_refresh
    self.async_update_listeners()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 178, in async_update_listeners
    update_callback()
    ~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 559, in _handle_coordinator_update
    self.async_write_ha_state()
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1019, in async_write_ha_state
    self._async_write_ha_state()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1141, in _async_write_ha_state
    self.__async_calculate_state()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1081, in __async_calculate_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1025, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/config/custom_components/panasonic_smart_app/sensor.py", line 576, in state
    _current_mode = list(
                    ~~~~~
        filter(lambda m: m[1] == int(mode), raw_mode_list)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    )[0][0]
    ~^^^
IndexError: list index out of range

**Comment by eoleedi** [comment]
要支援 NR-F551WX-W1 冰箱的話有一些問題需要解決

1. 溫度無法正常顯示
    ```
    2025-10-13 21:13:55.161 ERROR (MainThread) [custom_components.panasonic_smart_app] [冰箱-1 冷凍溫度] Unable to get temperature from status: {'0x00': '2', '0x01': '2', '0x57': '6', '0x50': '0', '0x0C': '0', '0x52': '0', '0x53': '0', '0x56': '255', '0x03': '-1683.3', '0x05': '-1683.3', '0x58': '-1683.3', '0x61': '-1683.3', '0x5A': '0', '0x5B': '0', '0x5C': '0'}
    Traceback (most recent call last):
      File "/config/custom_components/panasonic_smart_app/sensor.py", line 650, in native_value
        unsigned_temperature = int(status[self.command_type])
    ValueError: invalid literal for int() with base 10: '-1683.3'
    2025-10-13 21:13:55.167 ERROR (MainThread) [custom_components.panasonic_smart_app] [冰箱-1 冷藏溫度] Unable to get temperature from status: {'0x00': '2', '0x01': '2', '0x57': '6', '0x50': '0', '0x0C': '0', '0x52': '0', '0x53': '0', '0x56': '255', '0x03': '-1683.3', '0x05': '-1683.3', '0x58': '-1683.3', '0x61': '-1683.3', '0x5A': '0', '0x5B': '0', '0x5C': '0'}
    Traceback (most recent call last):
      File "/config/custom_components/panasonic_smart_app/sensor.py", line 650, in native_value
        unsigned_temperature = int(status[self.command_type])
    ValueError: invalid literal for int() with base 10: '-1683.3'
    2025-10-13 21:13:55.169 ERROR (MainThread) [custom_components.panasonic_smart_app] [冰箱-1 微凍結溫度] Unable to get temperature from status: {'0x00': '2', '0x01': '2', '0x57': '6', '0x50': '0', '0x0C': '0', '0x52': '0', '0x53': '0', '0x56': '255', '0x03': '-1683.3', '0x05': '-1683.3', '0x58': '-1683.3', '0x61': '-1683.3', '0x5A': '0', '0x5B': '0', '0x5C': '0'}
    Traceback (most recent call last):
      File "/config/custom_components/panasonic_smart_app/sensor.py", line 650, in native_value
        unsigned_temperature = int(status[self.command_type])
    ValueError: invalid literal for int() with base 10: '-1683.3'
    ```
2. 新鮮急凍結 RapidFreezingSensor無法顯示
    ```
    2025-10-13 21:13:55.170 ERROR (MainThread) [custom_components.panasonic_smart_app] Unknown value 255 for 冰箱-1 新鮮急凍結
    ```

    正常來說應該要是0或是828?
    
    ```
    "CommandType": "0x56",
    "CommandName": "新鮮急凍結",
    "ParameterType": "range",
    "ParameterUnit": "",
    "Parameters": [
        [
            "通常",
            0
        ],
        [
            "模式",
            828
        ]
    ]
    ```

3. 在調整冰箱強中弱的時候似乎也會觸發溫度的Bug(不確定原因)

**Comment by djurny** [comment]
Hi there,
Dehumidifier (+HEPA) F-YV28NH works.
- The device has no PM1.0 sensor so it always shows as `65535`.

Dehumidifier F-Y28CXW (with ~CZ-T004~ **CZ-T006** adapter) works. 
- The device has no PM1.0 nor PM2.5 sensors, so they always show as `65535`.
- The device has `nanoe` and not `nanoeX` function, yet the integration shows `nanoeX`. The panasonic app shows `nanoe` correctly and it seems that the device also reports `nanoe` instead of `nanoeX`:

  F-YV-28NH:
  ```
              {
                "CommandType": "0x0D",
                "CommandName": "nanoeX",
  ```
  
  F-Y28CXW:
  ```
              {
                "CommandType": "0x0D",
                "CommandName": "nanoe(清淨、脫臭)",
  ```
  Suspect that the class `PanasonicDehumidifierNanoeX(PanasonicBaseEntity, SwitchEntity)` assumes `0x0D` is always `nanoeX` and does not check if device has reported it as `nanoe`.

Groetjes,

**Comment by s900552000** [comment]
[home-assistant_panasonic_debug_2025-11-12T07-07-08.590Z.log](https://github.com/user-attachments/files/23494126/home-assistant_panasonic_debug_2025-11-12T07-07-08.590Z.log)
Device: Panasonic F-YV32NN  
ModelType: NNW-L  
DeviceType: 4 (Dehumidifier)  
Region: Taiwan (ems2.panasonic.com.tw)  
Integration version: 2.12.0  
Home Assistant Core: 2025.11.x  
Mode during capture: Continuous dehumidify (from Panasonic Smart App)

Observation:
- The device is successfully detected and added under the Panasonic Smart App integration.
- Only humidity sensor shows valid values in HA.
- Other entities (switch/select/humidifier) fail to initialize with IndexError (list index out of range).

Files attached: 
1. home-assistant_panasonic_debug_2025-11-12T07-07-08.590Z.log
(Each file is a direct copy from the integration debug log, with sensitive data such as `auth` and `cptoken` redacted.)
2. Panasonic app screenshot
3. HA dehumidifier feature screenshot
Thank you for reviewing this. Please let me know if more captures in other modes (e.g. Laundry Dry, Quiet) would help.
<img width="1179" height="2556" alt="F-YV32NN_APP_3" src="https://github.com/user-attachments/assets/afb53ba2-2c41-4d56-8efd-c805d3f79afe" />
<img width="1179" height="2556" alt="F-YV32NN_APP_2" src="https://github.com/user-attachments/assets/6b6dec40-98aa-45f2-aa23-dafa628e79ff" />
<img width="1179" height="2556" alt="F-YV32NN_APP_1" src="https://github.com/user-attachments/assets/d89dbc9a-4b0d-4810-a700-2a7518562388" />
<img width="415" height="753" alt="image" src="https://github.com/user-attachments/assets/f3ec505d-38fc-4511-bc63-b78a96e90955" />

**Reply by ryan1221tw-droid** [reply]
我的除濕機F-YV36NN也有遇到一樣的問題喔，在實體內找不到除濕機的開關，希望可以修正，謝謝

**Comment by alexruffell** [comment]
Any chance microwaves are or will be supported?

**Reply by osk2** [reply]
Which model are we talking about? I can't see any model with WiFi support on the official website.

**Reply by alexruffell** [reply]
I think they mostly claim Alexa control. This is one of them:

https://shop.panasonic.com/products/home-chef-connect-4-in-1-multi-oven

**Reply by osk2** [reply]
You should use [panasonic_cc](https://github.com/sockless-coding/panasonic_cc) instead. This component is intended for devices produced solely by Panasonic Taiwan.

**Comment by Quniwork** [comment]
回報，原本 F-Y20JH 正常使用，最近重新安裝後沒有顯示除溼機開關，希望能協助修正，謝謝

**Reply by osk2** [reply]
v2.14 已支援

**Comment by a0024** [comment]
<img width="1116" height="708" alt="image" src="https://github.com/user-attachments/assets/4078050f-64e1-4b0a-92e2-03fd164a63ea" />
請問VX系列的冷氣是否不支援

**Reply by a4leon** [reply]
CS-VX36FA2 也是以上畫面

**Reply by osk2** [reply]
v2.14 已支援

**Comment by jasonliaotw** [comment]
F-Y45CXW  with CZ-T004
無法正常連線，未來是否有支援的可能?
